### ISSUE TITLE
Protein mutation protocol implementation

### ISSUE THREAD

**Issue by ijpulidos** [OpenFreeEnergy/feflow#7]
Using the same base objects from the other protocols (`NonEquilibriumCyclingProtocol` and `RelativeHybridTopologyProtocol`) we want to implement a protocol that supports protein mutation based on the work from [@zhang-ivy using perses](https://github.com/choderalab/perses-barnase-barstar-paper).

**Comment by zhang-ivy** [comment]
The repo Iván linked is for protein mutations in protein:protein complexes. I use barnase:barstar to benchmark RBFEs and terminally-blocked amino acids to benchmark relative free energies (the latter was used to ensure everything was converging properly in the absence of sampling problems in protein protein interfaces)

We may also be interested in mutations in protein:small molecule complexes. @sukritsingh has been working with these systems, specifically kinase:inhibitor complexes, and should know the relevant systems to test here

**Comment by sukritsingh** [comment]
> We may also be interested in mutations in protein:small molecule complexes

Agreed! I think the "best" case to compare against would be something like Src or Abl kinase binding to the inhibitor Dasatinib. I have soon-to-be-published data on Abl-Dasatinib mutations that I can share, and there is previous work studying known Abl kinase mutations against dasatinib and imatinib we can draw upon (although those were done using FEP+). 

Relevant papers to some existing results that may be sufficient for your needs here: 
1. https://www.nature.com/articles/s42003-018-0075-x
2. https://pubs.acs.org/doi/full/10.1021/acscentsci.9b00590
3. https://pubs.acs.org/doi/full/10.1021/acscentsci.8b00717#
